sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/jit/ToRustJitLiteral.java

```diff
@@ -87,61 +101,61 @@ public VisitDecision preorder(DBSPDateLiteral node) {
         DBSPExpression expression = new DBSPConstructorExpression(
                 new DBSPPath("NaiveDate", "parse_from_str").toExpression(),
                 DBSPTypeAny.getDefault(), rustLiteral, new DBSPStrLiteral("%F"));
-        this.constant("Date", node, expression.unwrap());
+        this.constant(node.getType().code, node, expression.unwrap());
         return VisitDecision.STOP;
     }
 
     @Override
     public VisitDecision preorder(DBSPDecimalLiteral node) {
-        this.constant("Decimal", node);
+        this.constant(node.getType().code, node);
         return VisitDecision.STOP;
     }
 
     @Override
     public VisitDecision preorder(DBSPUSizeLiteral node) {
-         this.constant("Usize", node);
+         this.constant(node.getType().code, node);
          return VisitDecision.STOP;
     }
 
     @Override
     public VisitDecision preorder(DBSPISizeLiteral node) {
```

Comment:
Since most of the visit decisions are the same, you could use a default and only override the ones that differ. Maybe something like:

```
 @Override
    public VisitDecision preorder(DBSPLiteral node) {
        this.constant(node.getType().code, node);
        return VisitDecision.STOP;
    }

@Override
    public VisitDecision preorder(DBSPFloatLiteral node) {
        this.constant(node.getType().code, node.raw());
        return VisitDecision.STOP;
    }
    
    @Override
    public VisitDecision preorder(DBSPDoubleLiteral node) {
        this.constant(node.getType().code, node.raw());
        return VisitDecision.STOP;
    }
```

Reply:
unfortunately it doesn't work, since it calls the superclass preorder, which does something else.
the function this.preorder(DBSPLiteral) is never called.
I have to think whether I can fix this.